### ISSUE TITLE
When leaving out params-field in Websocket-RPC tm crashes

### ISSUE THREAD

**Issue by wolfposd** [tendermint/tendermint#708]
How to repoduce:

send `{"jsonrpc":"2.0","method":"dump_consensus_state","id":"0"}`

```
17:17:01 sent:  {"jsonrpc":"2.0","method":"dump_consensus_state","id":"0"}
17:17:01 system: Connection closed.Close status: 1006 (Abnormal Closure)Session length: 168714ms
```

tendermint crashes with following log:

```
github.com/tendermint/tendermint/rpc/lib/server.jsonParamsToArgsRPC(0xc4204cc4d0, 0x0, 0xc4207ee480, 0x14, 0xc420077220, 0x0, 0x0)
	/Volumes/HDD/wolf/golang/src/github.com/tendermint/tendermint/rpc/lib/server/handlers.go:210 +0x26
github.com/tendermint/tendermint/rpc/lib/server.(*wsConnection).readRoutine(0xc4202381b0)
	/Volumes/HDD/wolf/golang/src/github.com/tendermint/tendermint/rpc/lib/server/handlers.go:506 +0x7eb
created by github.com/tendermint/tendermint/rpc/lib/server.(*wsConnection).OnStart
	/Volumes/HDD/wolf/golang/src/github.com/tendermint/tendermint/rpc/lib/server/handlers.go:385 +0xab
```

still on tendermint 0.10.3 btw

**Comment by melekes** [comment]
```
{"jsonrpc":"2.0","method":"dump_consensus_state","id":"0","params":{}}
```
this works

**Comment by odeke-em** [comment]
@melekes your sample gives it the zero value whereas it omitting it entirely sets it as nil as at
https://play.golang.org/p/0qLcHQiX2w

Perhaps we should guard against this as on the failing line we do a dereference yet params is nil i.e.
https://github.com/tendermint/tendermint/blob/7682ad9a60162dd17fd6f61aeed7049a8635ac78/rpc/lib/server/handlers.go#L211-L213

and also guard against it in
https://github.com/tendermint/tendermint/blob/7682ad9a60162dd17fd6f61aeed7049a8635ac78/rpc/lib/server/handlers.go#L216-L222

**Comment by odeke-em** [comment]
This is actually on my list of things to fuzz soon :)

**Comment by melekes** [comment]
> Perhaps we should guard against this as on the failing line we do a dereference yet params is nil i.e.

no, we should not parse them in the first place (see https://github.com/tendermint/tendermint/commit/a3adac3787b5b52f64170d4e8925da18ffe70db9)

**Comment by odeke-em** [comment]
Nice!

**Comment by melekes** [comment]
```
λ curl http://localhost:46657/dump_consensus_state
{
  "jsonrpc": "2.0",
  "id": "",
  "result": {
    "round_state": {
      "Height": 1691,
      "Round": 0,
      "Step": 1,
      "StartTime": "2017-10-09T14:08:09.129491764+04:00",
      "CommitTime": "2017-10-09T14:08:08.129491764+04:00",
      "Validators": {
        "validators": [
          {
            "address": "EF243CC0E9B88D0161D24D733BDE9003518CEA27",
            "pub_key": {
              "type": "ed25519",
              "data": "2E0B9301334FCDAB193D514022F81BA09BBEC028685C96602BE9DD0BD4F9E202"
            },
            "voting_power": 10,
            "accum": 0
          }
        ],
        "proposer": {
          "address": "EF243CC0E9B88D0161D24D733BDE9003518CEA27",
          "pub_key": {
            "type": "ed25519",
            "data": "2E0B9301334FCDAB193D514022F81BA09BBEC028685C96602BE9DD0BD4F9E202"
          },
          "voting_power": 10,
          "accum": 0
        }
      },
      "Proposal": null,
      "ProposalBlock": null,
      "ProposalBlockParts": null,
      "LockedRound": 0,
      "LockedBlock": null,
      "LockedBlockParts": null,
      "Votes": {},
      "CommitRound": -1,
      "LastCommit": {},
      "LastValidators": {
        "validators": [
          {
            "address": "EF243CC0E9B88D0161D24D733BDE9003518CEA27",
            "pub_key": {
              "type": "ed25519",
              "data": "2E0B9301334FCDAB193D514022F81BA09BBEC028685C96602BE9DD0BD4F9E202"
            },
            "voting_power": 10,
            "accum": 0
          }
        ],
        "proposer": {
          "address": "EF243CC0E9B88D0161D24D733BDE9003518CEA27",
          "pub_key": {
            "type": "ed25519",
            "data": "2E0B9301334FCDAB193D514022F81BA09BBEC028685C96602BE9DD0BD4F9E202"
          },
          "voting_power": 10,
          "accum": 0
        }
      }
    },
    "peer_round_states": {
      "75EC8F15D244A421202F9725CD4DE509EE50303670310CF7530EF25E2B7C524B": {
        "Height": 1691,
        "Round": 0,
        "Step": 1,
        "StartTime": "2017-10-09T14:08:08.563251997+04:00",
        "Proposal": false,
        "ProposalBlockPartsHeader": {
          "total": 0,
          "hash": ""
        },
        "ProposalBlockParts": null,
        "ProposalPOLRound": -1,
        "ProposalPOL": null,
        "Prevotes": null,
        "Precommits": null,
        "LastCommitRound": 0,
        "LastCommit": null,
        "CatchupCommitRound": -1,
        "CatchupCommit": null
      }
    }
  }
}
```

Improved formatting.

**Comment by melekes** [comment]
Fixed in v0.11.1
